### PR TITLE
Add icestudio v0.5.0

### PR DESCRIPTION
**Pull request by qiwenmin**
After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, if **adding a new cask**:

- [x] Named the cask according to the [token reference].
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked there are no [open pull requests] for the same cask.
- [x] Checked the cask was not [already refused].
- [x] Checked the cask is submitted to [the correct repo].

[token reference]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/Homebrew/homebrew-cask/pulls
[already refused]: https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues
[the correct repo]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask

### CODE REVIEW

**Review by vitorgalvao — CHANGES_REQUESTED** [review]

**Comment by vitorgalvao** [comment]
Needs to be submitted to [homebrew/cask-drivers](https://github.com/Homebrew/homebrew-cask-drivers). We [define driver software](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask) as:

> Software to make a hardware peripheral recognisable and usable by the system. If the software is useless without the peripheral, it’s considered a driver.

**Comment by qiwenmin** [comment]
> Needs to be submitted to [homebrew/cask-drivers](https://github.com/Homebrew/homebrew-cask-drivers). We [define driver software](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask) as:
> 
> > Software to make a hardware peripheral recognisable and usable by the system. If the software is useless without the peripheral, it’s considered a driver.

Hello vitorgalvao,

Icestudio is not a driver for a hardware. It is an IDE for developing FPGA boards - is like Arduino for arduino boards. Arduino is submitted to homebrew-cask.

**Comment by vitorgalvao** [comment]
The cask still does need fixing, though.

**Comment by qiwenmin** [comment]
Thanks!

I will fix the homepage url and the comment soon.

> > Arduino is submitted to homebrew-cask.
> 
> Fair enough. Though that was also a close call, it‘s a precedent.
> 
> But the cask does still need fixing.

File: Casks/icestudio.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+cask 'icestudio' do
+  version '0.5.0'
+  sha256 '0968026bfb556605b7daa74dbe6bae0859cf3f8e7ab489bdfcf0d4c9ec3627cd'
+
+  url "https://github.com/FPGAwars/icestudio/releases/download/v#{version}/icestudio-#{version}-osx64.dmg"
+  appcast 'https://github.com/FPGAwars/icestudio/releases.atom'
+  name 'icestudio'
+  homepage 'https://github.com/FPGAwars/icestudio'
```
[the remark this answers is not in the document]
Comment (qiwenmin) [reply Casks/icestudio.rb]: The homepage is changed.

File: Casks/icestudio.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+cask 'icestudio' do
+  version '0.5.0'
+  sha256 '0968026bfb556605b7daa74dbe6bae0859cf3f8e7ab489bdfcf0d4c9ec3627cd'
+
+  url "https://github.com/FPGAwars/icestudio/releases/download/v#{version}/icestudio-#{version}-osx64.dmg"
```
[the remark this answers is not in the document]
Comment (qiwenmin) [reply Casks/icestudio.rb]: And the comment is added in the new commit.

**Review by vitorgalvao — APPROVED** [review]

**Pull request merged**